lib/grape/dsl/inside_route.rb

```diff
@@ -48,15 +48,20 @@ def declared_array(passed_params, options, declared_params, params_nested_path)
         end
 
         def declared_hash(passed_params, options, declared_params, params_nested_path)
+          renamed_params = route_setting(:renamed_params) || {}
+
           declared_params.each_with_object(passed_params.class.new) do |declared_param, memo|
             if declared_param.is_a?(Hash)
               declared_param.each_pair do |declared_parent_param, declared_children_params|
                 params_nested_path_dup = params_nested_path.dup
                 params_nested_path_dup << declared_parent_param.to_s
                 next unless options[:include_missing] || passed_params.key?(declared_parent_param)
 
+                rename_path = params_nested_path.dup + [declared_parent_param.to_s]
```

Comment:
Do we need to `.dup` it here? The `+` makes a copy I believe.

Reply:
You're right. This is a leftover of fiddling around and I wanted to be safe to not touch any state here. Fixed it.